gazebo_ros/src/gazebo_ros_properties.cpp

```diff
@@ -0,0 +1,603 @@
+  }
+  else
+  {
+    // get model parent name
+    gazebo::physics::ModelPtr parent_model = boost::dynamic_pointer_cast<gazebo::physics::Model>(model->GetParent());
+    if (parent_model) _res->parent_model_name = parent_model->GetName();
+
+    // get list of child bodies, geoms
+    _res->body_names.clear();
+    _res->geom_names.clear();
+    for (unsigned int i = 0 ; i < model->GetChildCount(); i ++)
+    {
+      gazebo::physics::LinkPtr body = boost::dynamic_pointer_cast<gazebo::physics::Link>(model->GetChild(i));
+      if (body)
+      {
+        _res->body_names.push_back(body->GetName());
+        // get list of geoms
+        for (unsigned int j = 0; j < body->GetChildCount() ; j++)
+        {
+          gazebo::physics::CollisionPtr geom = boost::dynamic_pointer_cast<gazebo::physics::Collision>(body->GetChild(j));
+          if (geom)
+            _res->geom_names.push_back(geom->GetName());
+        }
+      }
+    }
+
+    // get list of joints
+    _res->joint_names.clear();
+
+    gazebo::physics::Joint_V joints = model->GetJoints();
+    for (unsigned int i=0;i< joints.size(); i++)
+      _res->joint_names.push_back( joints[i]->GetName() );
+
+    // get children model names
+    _res->child_model_names.clear();
+    for (unsigned int j = 0; j < model->GetChildCount(); j++)
+    {
+      gazebo::physics::ModelPtr child_model = boost::dynamic_pointer_cast<gazebo::physics::Model>(model->GetChild(j));
+      if (child_model)
+        _res->child_model_names.push_back(child_model->GetName() );
+    }
+
+    // is model static
+    _res->is_static = model->IsStatic();
+
+    _res->success = true;
+    _res->status_message = "GetModelProperties: got properties";
+  }
+}
+
+void GazeboRosPropertiesPrivate::GetJointProperties(
+	gazebo_msgs::srv::GetJointProperties::Request::SharedPtr _req,
+	gazebo_msgs::srv::GetJointProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::JointPtr joint;
+  for (unsigned int i = 0; i < world_->ModelCount(); i ++)
+  {
+    joint = world_->ModelByIndex(i)->GetJoint(_req->joint_name);
+    if (joint) break;
+  }
+
+  if (!joint)
+  {
+    _res->success = false;
+    _res->status_message = "GetJointProperties: joint not found";
+  }
+  else
+  {
+    /// @todo: FIXME
+    _res->type = _res->REVOLUTE;
+
+    _res->damping.clear(); // to be added to gazebo
+    //_res->damping.push_back(joint->GetDamping(0));
+
+    _res->position.clear();
+    _res->position.push_back(joint->Position(0));
+
+    _res->rate.clear(); // use GetVelocity(i)
+    _res->rate.push_back(joint->GetVelocity(0));
+
+    _res->success = true;
+    _res->status_message = "GetJointProperties: got properties";
+  }
+}
+
+void GazeboRosPropertiesPrivate::GetLinkProperties(
+	gazebo_msgs::srv::GetLinkProperties::Request::SharedPtr _req,
+	gazebo_msgs::srv::GetLinkProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::LinkPtr body = boost::dynamic_pointer_cast<gazebo::physics::Link>(world_->EntityByName(_req->link_name));
```

Comment:
Regarding deboosting.. which would be a correct update for this line? Would the following be okay?
```cpp
auto body = std::shared_ptr<gazebo::physics::LinkPtr>(new world_->EntityByName(_req->link_name));
```
I guess this is not related to the usage of the new [conversion](https://github.com/ros-simulation/gazebo_ros_pkgs/tree/ros2/gazebo_ros/include/gazebo_ros/conversions) functions.

Reply:
This use of boost is coming from Gazebo, so it can't be removed. I added a [note](https://github.com/ros-simulation/gazebo_ros_pkgs/wiki/ROS-2-migration-contribution-guide/_compare/828dbbc519cd1bfcdb34d8d739d50a08c58a7920...6faaeed88f3ab2a70060a2434e1c3c58ee7f0874) to the tutorial.